### ISSUE TITLE
"Follow" command doesn't work after round restart

### ISSUE THREAD

**Issue by AtobaAzul** [Regalis11/Barotrauma#11180]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

AI crewmate's follow command stops working (greys out and they don't do anything) on map restart.

### Reproduction steps

1. Make an NPC follow you
2. Save the game
3. Retry

### Bug prevalence

Happens every time I play

### Version

Faction/endgame test branch

### Which operating system did you encounter this bug on?

Windows

**Comment by itchyOwl** [comment]
Fixed in https://github.com/Regalis11/Barotrauma-development/pull/3979

**Comment by Regalis11** [comment]
Seems to be fixed in single player, but doesn't work in MP.

**Comment by Rokvach** [comment]
Seems like the issue was with my build of the game, deleting the build and creating a new build fixed the issue for me. Working correctly. Closing.

**Comment by NilanthAnimosus** [comment]
Tested against Release/1.0 commit https://github.com/Regalis11/Barotrauma-development/commit/c81ad82458d0f6e3570943db493ed7bacba2c5e7

There still seemed to be potential issues with this in multiplayer even after cleaning my build by the follow order not following  the player correctly after a level transition or save reload.

notes this was a fresh save, where I deleted and rebuilt everything then put my config file back in (as well as serversettings and data folder) and still encountered the issue.

Judging by the code changes and the save file I extracted, its possible this is related to how it hashes the character info, and my specific combination of appearance, name and other details results in the order not working.

This is the player config element which I encountered the issue with:
```
<player
    name="Nilantht"
    tags="male,head1"
    HairIndex="10"
    BeardIndex="7"
    MoustacheIndex="0"
    FaceAttachmentIndex="6"
    HairColor="#A15837"
    FacialHairColor="#A15837"
    SkinColor="#F3C696">
    <job
      identifier="captain"
      variant="0" />
    <job
      identifier="medicaldoctor"
      variant="0" />
    <job
      identifier="securityofficer"
      variant="0" />
  </player>
```
